gapic-common/lib/gapic/common/retry_policy.rb

```diff
@@ -0,0 +1,187 @@
+        @initial_delay || DEFAULT_INITIAL_DELAY
+      end
+
+      def max_delay
+        @max_delay || DEFAULT_MAX_DELAY
+      end
+
+      def multiplier
+        @multiplier || DEFAULT_MULTIPLIER
+      end
+
+      def retry_codes
+        @retry_codes || DEFAULT_RETRY_CODES
+      end
+
+      def timeout
+        @timeout || DEFAULT_TIMEOUT
+      end
+
+      ##
+      # Perform delay if and only if retriable.
+      #
+      # Positional argument `error` takes precedence over positional arguments, meaning the latter
+      # will only be used if and only if `error` is not present.
+      #
+      # @return [Boolean] Whether the delay was executed.
+      #
+      def call error = nil, *args
+        should_retry = error.nil? ? retry?(*args) : retry_error?(error)
+        return false unless should_retry
+        perform_delay!
+      end
+      alias perform_delay call
+
+      ##
+      # Perform delay.
+      #
+      # @return [Boolean] Whether the delay was executed.
+      #
+      def perform_delay!
+        delay!
+        increment_delay!
+        true
+      end
+
+      ##
+      # Current delay value in seconds.
+      #
+      # @return [Numeric] Time delay in seconds.
+      #
+      def delay
+        @delay || initial_delay
+      end
+
+      ##
+      # @private
+      # @return [Boolean] Whether this error should be retried.
+      #
+      def retry_error? error
+        (defined?(::GRPC) && error.is_a?(::GRPC::BadStatus) && retry_codes.include?(error.code)) ||
+          (error.respond_to?(:response_status) &&
+            retry_codes.include?(ErrorCodes.grpc_error_for(error.response_status)))
+      end
+
+      ##
+      # @private
+      # Apply default values to the policy object. This does not replace user-provided values,
+      # it only overrides empty values.
+      #
+      # @param retry_policy [Hash] The policy for error retry. Keys must match the arguments for
+      #   {Gapic::Common::RetryPolicy.new}.
+      #
+      def apply_defaults retry_policy
+        return unless retry_policy.is_a? Hash
+        @retry_codes   ||= convert_codes retry_policy[:retry_codes]
+        @initial_delay ||= retry_policy[:initial_delay]
+        @multiplier    ||= retry_policy[:multiplier]
+        @max_delay     ||= retry_policy[:max_delay]
+
+        self
+      end
+
+      # @private Equality test
+      def eql? other
+        other.is_a?(RetryPolicy) &&
+          other.initial_delay == initial_delay &&
+          other.max_delay == max_delay &&
+          other.multiplier == multiplier &&
+          other.retry_codes == retry_codes &&
+          other.timeout == timeout
+      end
+      alias == eql?
+
+      # @private Hash code
+      def hash
+        [initial_delay, max_delay, multiplier, retry_codes, timeout].hash
+      end
+
+
+      private
+
+      def retry? *args
+        args.empty? ? retry_with_deadline? : retry_error?(*args.first)
+      end
+
+      def delay!
+        Kernel.sleep delay
```

Comment:
Just say

```ruby
sleep delay
```

The Kernel module is mixed into the Object base class, so its methods are always directly available.

Reply:
We currently rely on `Kernel.stub` for some of our former tests (e.g. [gapic-common/test/gapic/generic_lro_test.rb](https://github.com/googleapis/gapic-generator-ruby/blob/27da6f7bab8d011aa64d96ea8109c3093c82d0f0/gapic-common/test/gapic/generic_lro_test.rb#L266). 

Switching to `sleep` breaks that and will need further tweaking. If you feel strongly about this change, please send it back.

Reply:
Hmm. Okay, not that important.